port_ocean/cache/disk.py

```diff
@@ -0,0 +1,47 @@
+import pickle
+from pathlib import Path
+from typing import Any, Optional
+
+from port_ocean.cache.base import CacheProvider
+
+
+class DiskCacheProvider(CacheProvider):
+    STORAGE_TYPE = "disk"
+
+    def __init__(self, cache_dir: str | None = None) -> None:
+        if cache_dir is None:
+            cache_dir = ".ocean_cache"
+        self._cache_dir = Path(cache_dir)
+        self._cache_dir.mkdir(parents=True, exist_ok=True)
+
+    def _get_cache_path(self, key: str) -> Path:
+        return self._cache_dir / f"{key}.pkl"
+
+    async def get(self, key: str) -> Optional[Any]:
+        cache_path = self._get_cache_path(key)
+        if not cache_path.exists():
+            return None
+
+        try:
+            with open(cache_path, "rb") as f:
+                return pickle.load(f)
+        except (pickle.PickleError, EOFError):
+            raise RuntimeError(f"Failed to read cache file: {cache_path}")
+
+    async def set(self, key: str, value: Any) -> None:
+        cache_path = self._get_cache_path(key)
+        try:
+            with open(cache_path, "wb") as f:
+                pickle.dump(value, f)
+        except (pickle.PickleError, IOError) as e:
+            raise RuntimeError(f"Failed to write cache file: {e}")
+
```

Comment:
same here, I am not sure if we want to just fail the whole resync process due to some issue in caching ( I am okay with it being raised here, but we need to make sure we are handling it in the `utils/cache.py` handler